pkg/postgres/upserter.go

```diff
@@ -43,7 +43,9 @@ func newRunRequired(
 
 //AddRecordToDatabase upserts a record to the database, while
 //attempting to reconcile separate records from the same train run
-func (a *UpserterAgent) AddRecordToDatabase(rec martaapi.Schedule) (err error) {
+func (a *UpserterAgent) AddRecordToDatabase(recs []martaapi.Schedule, i int) (err error) {
```

Comment:
super nitpick: could we name `i` more descriptively here? Note, the spiel below is non-blocking, but just trying to articulate some confusion I was feeling while reviewing.  This could also just be my unfamiliarity with the code in general though!

This code I found a bit confusing, but I think I'm slowly starting to understand.  We pass in the entire run, and an index for the specific event in the run that we want to insert.  We need the entire run in order to attempt to correct the line and station if the specific event is incorrect -- that is, if we had a "southbound event" in the middle of a "northbound" run, we'd want to correct that event. 

I'm wondering if it would be possible to separate out that behavior from the `upserter` agent, such that `AddRecordToDatabase` didn't do any sort of reconciliation, but that business logic was handled in the `Dumper` instead? `AddRecordToDatabase` would just handle the database interactions, and the `dumper` would control any pruning/parsing/business logic to ensure that the runs are formatted how we want them?  I was often referencing this code back to the dumper implementation in order to understand what was going on.

Reply:
I chose to put it here for basically "performance" reasons, but the impact is probably so incredibly tiny that I definitely see how it might make more sense to do this upstairs. Basically, there's no reason to do correction if we're adding new estimate to an existing run, it's only necessary if we're creating a new run, and the only way to tell whether a run is new or not is by interacting with the DB.

The actual correction code (while big and sort of ugly) is pretty dang simple though. It's just looping a couple times over something that should never contain more than like 15 entires, and I think it's really not worth the mixing-of-concerns to save a handful of processor cycles 👍

One thing to keep in mind though is that the other dumpers won't be capable of correction to the same extent that the PG dumper is - the strength of PG is that, even at the very end of a run when there's only one station left, we'll still link it up to the information we had earlier about the run, which includes an already corrected direction. We can re-use the corrector on any of the dumpers, but they'll get less reliable as the train nears the end of its line!